src/__tests__/linter.karma.test.ts

```diff
@@ -0,0 +1,49 @@
+import { FetchMockSandbox } from 'fetch-mock';
+import { Spectral } from '../spectral';
+
+describe('Linter', () => {
+  let fetchMock: FetchMockSandbox;
+  let spectral: Spectral;
+
+  beforeEach(() => {
+    fetchMock = require('fetch-mock').sandbox();
+    window.fetch = fetchMock;
+    spectral = new Spectral();
+  });
+
+  afterEach(() => {
+    window.fetch = fetch;
+  });
+
+  describe('custom functions', () => {
+    it('should be able to make a request using fetch', async () => {
+      fetchMock.mock('https://stoplight.io', {
```

Comment:
@P0lip Just off-topic thought: This current test set up compels us to duplicate tests _and_ use two different libs to intercept fetch calls _and_, depending on the the scope, use 3 different suffixes for test files (jest, karma or void). 

This makes contributing and reviewing a little bit harder than it should be.

I feel there might be a better way to do this.

Thoughts?

Reply:
> @P0lip Just off-topic thought: This current test set up compels us to duplicate tests and use two different libs to intercept fetch calls and, depending on the the scope, use 3 different suffixes for test files (jest, karma or void).

Yeah, that is true.

> I feel there might be a better way to do this.

We could simply drop nock and use fetch-mock everywhere, I believe.
Nock is pretty cool though as it's an actual server.

Reply:
@P0lip I'm going to log an issue about this to keep track of it.